compiler/qsc_frontend/src/resolve.rs

```diff
@@ -881,29 +897,42 @@ fn bind_global_item(
     namespace: &Rc<str>,
     next_id: impl FnOnce() -> ItemId,
     item: &ast::Item,
-) -> Result<(), Error> {
+) -> Result<(), Vec<Error>> {
     match &*item.kind {
         ast::ItemKind::Callable(decl) => {
             let item_id = next_id();
             let status = ItemStatus::from_attrs(&ast_attrs_as_hir_attrs(item.attrs.as_ref()));
             let res = Res::Item(item_id, status);
             names.insert(decl.name.id, res);
+            let mut errors = Vec::new();
             match scope
                 .terms
                 .entry(Rc::clone(namespace))
                 .or_default()
                 .entry(Rc::clone(&decl.name.name))
             {
-                Entry::Occupied(_) => Err(Error::Duplicate(
+                Entry::Occupied(_) => errors.push(Error::Duplicate(
```

Comment:
Why accumulate these two errors instead of just returning the first one? If we already saw the name is duplicated, is it worth _also_ checking for duplicate intrinsic?

Reply:
I had to think about this for a while, as I forgot what my motivation was when I did the update this way. It turns out that a callable can be both a name collision (if the fully qualified namespace.name is a direct match) and an intrinsic collision (if declared as `body intrinsic` and another intrinsic in ANY namespace has the same name). This can be true even in the case where the duplicate intrinsic is an internal from a distinct namespace. I didn't want either error to mask the other, so I wanted to make sure I could report both.

Reply:
I added a test case to show that both errors can be returned if the callable name violates each rule separately.